packages/app/pages/index.tsx

```diff
@@ -201,31 +230,312 @@ const Tab: React.FC<{ to: string }> = ({ children, to }) => {
+                      reader.readAsDataURL(file);
+                    }
+                  }
+                }}
+              />
+            </Label>
+            {fileLoading && "Loading..."}
+          </>
+        )}
+        {stage === 2 && (
+          <>
+            <h2>Additional Metadata</h2>
+            {additionalFields.map((a, i) => (
+              <AdditionalFieldRow key={i}>
+                <ModalInput
+                  label={"Key"}
+                  value={a.key}
+                  onChange={(e) =>
+                    setAdditionalFields(
+                      additionalFields.map((field, j) =>
+                        j === i
+                          ? { value: field.value, key: e.target.value }
+                          : field
+                      )
+                    )
+                  }
+                />
+                <ModalInput
+                  label={"Value"}
+                  value={a.value}
+                  onChange={(e) =>
+                    setAdditionalFields(
+                      additionalFields.map((field, j) =>
+                        j === i
+                          ? { value: e.target.value, key: field.key }
+                          : field
+                      )
+                    )
+                  }
+                />
+                <Button
+                  leftIcon={<TrashIcon />}
+                  onClick={() =>
+                    setAdditionalFields([
+                      ...additionalFields,
+                      { key: "", value: "" },
+                    ])
+                  }
+                />
+              </AdditionalFieldRow>
+            ))}
+            <Button
+              onClick={() =>
+                setAdditionalFields([
+                  ...additionalFields,
+                  { key: "", value: "" },
+                ])
+              }
+            >
+              Add Field
+            </Button>
+          </>
+        )}
+        {stage === 3 && (
+          <>
+            <h2>Details</h2>
+            <p>
+              <b>Name:</b> {name}
+            </p>
+            <p>
+              <b>Symbol:</b> {symbol}
+            </p>
+            <p>
+              <b>Quantity:</b> {quantity}
+            </p>
+            <p>
+              <b>Price:</b> {price} ETH
+            </p>
+            <p>
+              <b>Thumbnail:</b>
+            </p>
+            <IpfsImage cid={cid} />
+            {additionalFields.map((a) => (
+              <p key={a.key}>
+                <b>{a.key}:</b> {a.value}
+              </p>
+            ))}
+          </>
+        )}
+      </Modal>
+    </>
+  );
+};
+
+const MembershipTabContent = () => {
+  const [memberships, setMemberships] = useState<IMembershipProps[]>([]);
+  const address = useAddress();
   const web3 = useWeb3();
   const chainId = useChainId();
+  const [redirectionUrls, setRedirectionUrls] = useState<{
+    [key: string]: string | undefined;
+  }>({});
```

Comment:
We might be able to add the redirect urls to the passport metadata along with thumbnail instead of firebase

Reply:
That's a great point, forgot that editing the JSON would incur new hashes that would cost gas